### PR TITLE
refactor(x/evm): Break host object out of keeper and cleanup.

### PR DESCRIPTION
**Pull request by BrickBera**

### CODE REVIEW

File: cosmos/x/evm/keeper/host/host.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	ethTxMempool sdkmempool.Mempool,
+	offChainKv *offchain.Store,
+	rpcProvider evmrpc.Provider,
+) *Host {
+	// We setup the host with some Cosmos standard sauce.
+	h := &Host{}
+
+	// Build the Plugins
+	h.bp = block.NewPlugin(storeKey)
+	h.Cp = configuration.NewPlugin(storeKey)
```
Comment (BrickBera) [inline cosmos/x/evm/keeper/host/host.go:76]: have to make this public

File: cosmos/x/evm/keeper/msg_server.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func (k *Keeper) UpdateParams(
 	}
 
 	// Update the params.
-	k.cp.Prepare(ctx)
-	k.cp.SetParams(&req.Params)
+	k.Host.Cp.Prepare(ctx)
```
Comment (BrickBera) [inline cosmos/x/evm/keeper/msg_server.go:74]: Cp become public because of this usage

File: cosmos/x/evm/keeper/host/host.go (hunk 1 above)
Comment (itsdevbear) [reply cosmos/x/evm/keeper/host/host.go:76]: Is there a way we can keep it private?

File: cosmos/x/evm/keeper/msg_server.go (hunk 2 above)
Comment (itsdevbear) [reply cosmos/x/evm/keeper/msg_server.go:74]: I think it would be nice if we could keep it private

File: cosmos/x/evm/keeper/host.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	// We setup the host with some Cosmos standard sauce.
+	h := &Host{}
+
+	// Build the Plugins
+	h.bp = block.NewPlugin(storeKey)
+	h.cp = configuration.NewPlugin(storeKey)
+	h.gp = gas.NewPlugin()
+	h.hp = historical.NewPlugin(h.bp, offChainKv, storeKey)
+	h.txp = txpool.NewPlugin(h.cp, rpcProvider, utils.MustGetAs[*mempool.EthTxPool](ethTxMempool))
+	return h
```
Comment (itsdevbear) [inline cosmos/x/evm/keeper/host.go:80]: return &Host{
	bp: block.NewPlugin(....),
	....
}

File: cosmos/x/evm/keeper/host.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+// GetTxPoolPlugin returns the txpool plugin.
+func (h *Host) GetTxPoolPlugin() core.TxPoolPlugin {
+	return h.txp
+}
+
+// GetAllPlugins returns all the plugins.
+func (h *Host) GetAllPlugins() []plugins.BaseCosmosPolaris {
+	return []plugins.BaseCosmosPolaris{h.hp, h.cp, h.gp, h.pp, h.sp}
```
Comment (itsdevbear) [inline cosmos/x/evm/keeper/host.go:137]: missing some plugins here (I think this was from before your pr tho), also "hp" should be "bp"

File: cosmos/x/evm/keeper/keeper.go [hunk 5]
```diff
@@ -63,15 +50,8 @@ type Keeper struct {
...
-	// The various plugins that are are used to implement `core.PolarisHostChain`.
-	bp  block.Plugin
-	cp  configuration.Plugin
-	gp  gas.Plugin
-	hp  historical.Plugin
-	pp  precompile.Plugin
-	sp  state.Plugin
-	txp txpool.Plugin
+	// The host contains various plugins that are are used to implement `core.PolarisHostChain`.
+	Host *Host
```
Comment (itsdevbear) [inline cosmos/x/evm/keeper/keeper.go:54]: we can probably make Host private tbh, both the variable and the type.

File: cosmos/x/evm/keeper/processor_test.go [hunk 6]
```diff
@@ -142,9 +142,9 @@ var _ = Describe("Processor", func() {
 			tx := coretypes.MustSignNewTx(key, signer, legacyTxData)
 			addr, err := signer.Sender(tx)
 			Expect(err).ToNot(HaveOccurred())
-			k.GetStatePlugin().CreateAccount(addr)
-			k.GetStatePlugin().AddBalance(addr, big.NewInt(1000000000))
-			k.GetStatePlugin().Finalize()
+			k.Host.GetStatePlugin().CreateAccount(addr)
+			k.Host.GetStatePlugin().AddBalance(addr, big.NewInt(1000000000))
```
Comment (itsdevbear) [inline cosmos/x/evm/keeper/processor_test.go:146]: yeah make host private and provide an accessor.

File: cosmos/x/evm/keeper/host.go (hunk 3 above)
Comment (BrickBera) [reply cosmos/x/evm/keeper/host.go:80]: Hard to do so, because `h.hp = historical.NewPlugin(h.bp, offChainKv, storeKey)` requires h.bp and `h.txp = txpool.NewPlugin(h.cp, rpcProvider, utils.MustGetAs[*mempool.EthTxPool](ethTxMempool))` requires h.cp

File: cosmos/x/evm/keeper/host.go [hunk 7]
```diff
@@ -46,6 +46,18 @@ import (
 // Compile-time interface assertion.
 var _ core.PolarisHostChain = (*host)(nil)
 
+type Host interface {
+	core.PolarisHostChain
+	GetAllPlugins() []plugins.BaseCosmosPolaris
```
Comment (itsdevbear) [inline cosmos/x/evm/keeper/host.go:51]: can we add comments as to why we export these?

File: cosmos/x/evm/plugins/configuration/plugin.go [hunk 8]
```diff
@@ -37,8 +37,8 @@ import (
 
 // Plugin is the interface that must be implemented by the plugin.
 type Plugin interface {
-	plugins.BaseCosmosPolaris
 	core.ConfigurationPlugin
+	plugins.BaseCosmosPolaris
```
Comment (itsdevbear) [inline cosmos/x/evm/plugins/configuration/plugin.go:41]: revert this change, this was me my bad :D

File: cosmos/x/evm/keeper/host.go (hunk 3 above)
Comment (itsdevbear) [reply cosmos/x/evm/keeper/host.go:92]: thats true, good point.
